src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,99 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_LIST_SIZE = 10;
+    private T[] values;
+    private int currentListSize;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_LIST_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        values[currentListSize++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkAdditionRange(index);
+        growIfArrayFull();
+        System.arraycopy(values, index, values, index + 1, currentListSize - index);
+        values[index] = value;
+        currentListSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
```

Comment:
could be replaced with foreach

Reply:
agreed in slack chat to stay here with for, becase for each will require iterator overriding